### PR TITLE
Fix xDS logic for synthetic PeerAuthentications

### LINKED ISSUE #46570: Port level mTLS mode not rejecting out-of-mesh traffic
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

In a setup where you have a helloworld service running in a `test` namespace with the ambient label I applied the following PeerAuthentication to that namespace:
```yaml
apiVersion: security.istio.io/v1beta1
kind: PeerAuthentication
metadata:
  name: helloworld-strict
  namespace: test
spec:
  mtls:
    mode: PERMISSIVE
  portLevelMtls:
    "5000":
      mode: STRICT
  selector:
    matchLabels:
      app: helloworld
```
I then tries to call the helloworld service from another namespace that is out of the mesh (not labeled).
Currently requests hits the service and response received.
I tested it with sidecars and it rejects the traffic on that port with the same config.

@keithmattix Is this supposed to be supported?

### Version

```Text
Current master.
```

### Additional Information

_No response_

### Affected product area

- [X] Ambient
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [ ] Upgrade
- [ ] Multi Cluster
- [ ] Virtual Machine
- [ ] Control Plane Revisions

### PR DESCRIPTION
**Pull request by keithmattix**
**Please provide a description of this PR:**
Delta xDS requests were asking for the synthetic PeerAuthentication policy, but our RBAC xDS handlers was telling ztunnel to remove it because it was expecting the actual PeerAuthentication's resource (since that's the configuration that changed/created)

Fixes #46570

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [X] Ambient

### CODE REVIEW

File: pilot/pkg/xds/workload.go [hunk 1]
```diff
@@ -151,18 +152,27 @@ func (e WorkloadRBACGenerator) GenerateDeltas(
...
+				Kind:      p.Kind,
+			})
+		}
 	}
 	if len(req.ConfigsUpdated) != 0 && len(updatedPolicies) == 0 {
 		// This was a incremental push for a resource we don't watch... skip
 		return nil, nil, model.DefaultXdsLogDetails, false, nil
 	}
+
 	policies := e.s.Env.ServiceDiscovery.Policies(updatedPolicies)
```
Comment (howardjohn) [inline pilot/pkg/xds/workload.go:170]: seems like the "convert to syntethic ones" should be handled in Policies()? otherwise we are kind of leaking a bit

Comment (howardjohn) [reply pilot/pkg/xds/workload.go:170]: maybe not actually since we need to know them for the expected/removed stuff as well..

**Review by howardjohn — APPROVED** [review]

File: pilot/pkg/xds/workload.go (hunk 1 above)
Comment (keithmattix) [reply pilot/pkg/xds/workload.go:170]: Exactly, that was the crux of the issue. The diffing was off so ztunnel removed the policy from its local state, leading to the policies being ignored

Comment (howardjohn) [reply pilot/pkg/xds/workload.go:170]: The real root cause of the leaking abstraction is the ConfigUpdate being pretty lose with what events we send. We do the same in other places though, not so bad just a bit tricky for cases like this.

Comment (keithmattix) [reply pilot/pkg/xds/workload.go:170]: Yeah it looks like PeerAuthentication gets handled in the generic `configHandler` [here](https://github.com/istio/istio/blob/master/pilot/pkg/bootstrap/server.go#L882). I can add a TODO to split PeerAuthentication out with its own logic if you think it'll be helpful

**Pull request merged**